### PR TITLE
Adds an incentive to play security

### PR DESCRIPTION
**Pull request by GuppyLaxx**
People in security now begin with martial arts, HoS gets CQC, warden gets Krav Maga, removing the gloves in his locker, officers get 1 out of 6 small martial arts randomly.
### **Officer martial arts :**
**Cool Flips Bro** : Flip attacks.
**Block Affinity** : You're prone to preemptively blocking.
**Heavy Hitter** : You beat the shit out of people.
**Hard Headed** : You lack a couple brain cells and usually headbutt people.
**Fierce Grabber** : Your grip is pretty strong. 🍆 
**Tactical Disarmer** : You yank those weapons right out.
*Figure out what they do better by playing or being a lame shit and looking at the code.*

Detective gets the most OP gun on station so go whine at someone else if you want him getting a martial art.

### CODE REVIEW

**Comment by MacHac** [comment]
Most OP gun on the station?  Since when does the detective start with a taser?

Also, security doesn't need a buff.  Robust officers can already crush group antags on their own.  If it DID need a buff, then that buff should be consistent, not just throwing more RNG on the pile.

**Comment by GuppyLaxx** [comment]
`a taser`
So you're one of those, ain't ya? Considering how just about every single fucking antag in this game has a counter to those, it seems you might have started with the "Hard Headed" trait this time.
Who wins, an officer with his tools or 2 greytiders with literally the most ghetto gear?
Maybe with that mentality the question will go over your head and say the officer does.
And this isn't some `haha straight up buff lmao!!!`, yes, it's chance based for a fucking reason, you get your ass saved every once in a while, not constantly. It's almost like that's the whole purpose?

**Comment by ghost** [comment]
While interesting I'm pretty against this one.

**Comment by Rayisken** [comment]
Give the Det a grappling hook instead https://youtu.be/TYaF-lQlhAU?t=23s

**Comment by MacHac** [comment]
It is by definition a straight up buff since there are no drawbacks to it ever.  It is inconsistent, but it is still a buff.

Also, officers carry a counter to adrenaline glands/implants.  Its called a taser.  You tase them again, and you're back to where you started.  Or, even better, you close to melee range and baton.  The instant the perp runs out of charges on their 'get-out-of-shitty-stun-combat-free card', assuming they didn't have anything better to spend 8 TC/one gene point on, you win.  If your taser runs out of charges, you still have at LEAST one flashbang, plus an EMP-resistant baton.

The detective's gun is better when under threat of EMPs; However, flashbangs followed by a baton to the face work even better than that, and with the advantage of not needing to aim.  I'm not saying the det's gun isn't good, or even that it's not OP.  I'm saying that in most actual use cases, you're better off with a taser.  For example, in the officer vs two greytiders example, unless the greytiders are already standing next to the officer he can tase them both before they can close to melee range (assuming all parties involved are gud@gaem and don't misclick.)  If two greytiders ARE standing next to the officer, that's the officer's fuckup, because experienced officers don't stand next to greytiders.

Tangent aside, we're not talking about taser balance here.  We're talking about 'does security need a buff', and the answer is fuck no.  If and when Deadlybagofdoom and the like stop being able to solo arrest entire cults with a box of flashbangs and two tasers, it might be a good idea to look at a *consistent* security buff.  Until that day comes, if anything security needs a nerf.

**Comment by GITGUDHIHIHIHUEHUEHUEEEEE** [comment]
It's great! That would bring officers the edge they are supposed to have in confrontations, or can any stray kid with a toolbox clubber a police officer? No, and that's because they have superb training (amongst other things). Really, I think they should have higher knockdown resistances and by default krav-maga, or any martial art, learned from the start. That would make security officers much, much more robust and better prepared to take down extremely misguided citizens.

**Comment by ghost** [comment]
@GITGUDHIHIHIHUEHUEHUEEEEE kill yourself

**Comment by kayozz** [comment]
Whilst I disagree with everything Adam ever stands for, he's correct in this being a terrible idea.

**Comment by Nirnael** [comment]
S P E E D M E R G E

**Comment by GITGUDHIHIHIHUEHUEHUEEEEE** [comment]
You first adam.

**Comment by CMOisLing** [comment]
While interesting, this is a bad idea for reasons stated above. But this could be used later for a curator rework so they have something better to do than print wgw copies

**Comment by epicmemename** [comment]
@MacHac I am thoroughly convinced you either do not play this game or are engaging in complete satire. I'm inclined to believe the latter because even if you don't play the game, you can still read the code, and someone who has even briefly read the code would see that in addition to removing all stuns, activating the implant gives you both synaptzine and "stimulants", both very good at dealing with stuns (combined it's 8 seconds off a stun every reaction, while taser is a 10 second stun). 

How many times can you tase someone with a taser? 5 times. How many times can you stun someone with a detectives gun? Depends on how much ammo you made. 

You are not "back where you started", you are under constant pressure of the chems and the fact that the "perp" can use his adrenals again (assuming he has not used them before). 

FYI, keep the argument fair, bringing in items like batons and flashbangs doesn't count considering both of the jobs can have access to those resources. Even in the case of flashbanging someone then batoning them, flashbangs are intentionally designed to be countered by flash protect and ear protection, and the adrenals can be used right after you baton them, and they can run away before you can hit them again. 

In the case of using the det's gun against someone with adrenals, you can spam the gun to stun them over and over with little problem, seeing as you can reload it in an instant. This, added to the fact that there's a much better range for dets bullets compared to tasers, the fact that they do real damage and not just stun, makes it a much better choice for actual, real situations in game. 

The last part is entirely anecdotal. You can deal easily with bad players regardless of circumstance.

**Comment by ghost** [comment]
i am thoroughly convinced everyone in this PR is a fucking retard

no amount of the usual 'but stuns are strong' 'yeah well adrenals exist' 'ok but what if you don't have adrenals' 'get good or get meth' nonsense matters whatsoever because it's a completely circular debate and the fact of the matter is that you can't account for any given situation in a github comment

except me, because i am a level 15 psionic, but that doesn't matter

in any case the problem with this PR is that it won't accomplish what it intends to do because if people wanted to play sec for the cool loot they fucking would. they have all the cool loot ever. what they don't have is a reason to play sec because it's a tiring and thankless job where you get bumped off by every joe, dick, and harry because your team is never useful and then get whined at by admins for scratching your balls. 

in fact, attempting to attract people to play sec by giving them cool shit will attract the exact opposite of sec players that you want, so it's frankly just going to be detrimental if it has a real effect at all.

that being said, yes, there are balance concerns in undisarmable melee stuns for the same reason people didn't like stun gloves that much and were iffy about krav maga. krav maga is actually pretty damn strong but it's in the form of gloves and honestly it's tolerable with cooldowns and all. martial arts don't really have that and there's a reason they're somewhat rare or traitor items, because in the hands of a robust player they're really fuckin' strong. 

especially CQC. CQC's a nukie item for a reason and before you say 'chef in his kitchen also has it' need i remind you of the meme days when the chef wasn't restricted to his kitchen with his close quarters cooking and how being in the kitchen with an angry chef is virtually an instant death zone as it is now?

it's got a two hit long time knockout and that's very powerful. yes, you can make the argument that stunbaton is better or so forth or that guns exist and yes, i'll agree with you there, but there's a reason weapons other than only stunbatons and so forth are used, and that's because while powerful they aren't literally the only weapon ever worth using as people like to claim. and also, stunbatons are disarmable-that's why melee weapons are risky. if you can push the stunbaton out of the guy's hand, you can then hit him with it. the difference with CQC is that you can't do that-it's not disarmable and the guy can outright counter your melee attacks if he has throwmode on.

being able to knock people out immediately is POWERFUL. i don't need to elaborate on that, but i will, because i like talking. preventing people from being able to call their buddies to X location, being able to take down a crowd with a flashbang and then rapidly neckchopping everyone hit so they don't get up for a minute or so-that's pretty goddamn powerful, and while it won't make a lick of difference in the hands of most security players because they're retarded it'll only take one half competent player to point out that it's some seriously powerful, dangerous stuff. 

i don't feel like CQC should be in the list of rotated martial arts at all, if you feel like you need to add this.

but to continue on that vein, i still need to point out that undisarmable melee stuns are bad. it gives the other person a very very significant edge in a situation where normally they would be punished for being a fucking moron and not having their stunbaton or running out of charge on all their weapons or being EMPd or so forth.

i mean, obviously, for one, it's going to make it impossible for changelings to impersonate security staff, because 'PROVE YOU ARE SEC BY KARATE CHOPPING ME'. but it also means that security has a given edge over anyone else by default even if otherwise debilitated and stark naked, and while not a significant balance concern that's only because yogs is fucking moronic. if they weren't, this would honestly be a pretty big damn issue because if you stun two sec officers and grab both their weapons and kill one and severely damage the other, it means that you can't close in on melee to finish one off without running a very significant risk of getting blasted down asshole avenue by paul blart, space gestapo.

is that good? well, honestly, i don't think so. not in the vein of 'it'll make it too punishing for antags' because, again, yog is retarded. it won't make too much of a difference if sec has pulse rifles or nerf guns because either way the traitor and security will both be too busy ahelping 'hOW DO I UNBCKLE' to start shooting at each other. but in terms of encouraging snowflakiness, yes, that's bad. it's going to lead into things like 'shouldn't X job be faster at X than Y' or so forth because people are ALREADY using the argument that sec should by default be better at fighting than anyone else because it makes sense and it's logical and honestly the moment you say the word 'logic' in SS13 i hope that your proctologist is cursed by witch doctor and anything he touches on your body turns into an anus.

it's just going to encourage people to make similar PRs like 'engineers can weld faster than other people' or 'doctors can apply brute packs faster than other people or do surgery faster' and that's a spiral hole i don't need to explain why that's bad. but it also goes against a core crux of SS13-that anyone with sufficient competence can beat out anyone, because people don't have innate advantages over each other and the only balancing factors are being smart and gear. and you can't remove this gear. it's innate. i've said that before. it's undisarmable. it's not gloves or gear granted like krav maga is so you can't steal it for yourself and turn the tables on sec, you can't impersonate sec with it, and you can't cut off an arm to render the gloves useless.

it's also just kind of memey, but that's personal opinion.

tl;dr it's bad because it encourages snowflakiness in jobs, it's bad because it's undisarmable, and it's bad because it gives people an innate advantage over other people they really shouldn't have if they manage to lose their weapons. it's not a serious balance concern, but it sets the road for future bullshit and the actual serious balance concern is CQC being in the rotation at all, which is hilariously strong. if you insist on adding it, take out CQC and please, for christ sake, make it an item that sec officers wear, like gloves, rather than an innate martial art which can't be impersonated by changelings.

**Comment by ghost** [comment]
oh also aggressive grabber is stupid. seriously, it's just stunbatons for hands. you grab the person and then they're stunned. it's one of the two reasons sleeping carp is strong, the other being gun immunity.

**Comment by GuppyLaxx** [comment]
Aggro grabs aren't even that. Not even close. Relying on aggro grabs are just gonna get you fucked in the ass harder than stunbatoning a prisoner from the bar to the brig then throwing them in a cell.
Krav maga had zero cooldowns before, and still does, what change does fucking giving it as a martial art rather than some shitty gloves that you have to pay for using in the long run because `haHA not insulated!!!!`, you might aswell just keep your harm baton out instead of using that garbage.
Your screaming consits of attempting to shit on CQC while admitting what literally shits on it in the first line of it, ranged weapons. Amazing how CQC doesn't change that, ain't it??? Also CQC can only counter batons, and no other melee weapons.
You go on about "undisarmable stuns", one of the common martial arts is a 50% chance to trigger, and THEN a 20% chance to `stun for 2 seconds and that's it`, you're not only relying on certain RNG, but also fucking yourself in the ass by going about with your arms flailing
The other "undisarmable stun" is headbutting which is `countered by wearing a helmet, and that's it, easy as that`
Flipper is stupid and dumb for a reason, and still doesn't change shit, literally "ranged attacks" applies to every single one of these martial arts, thrown shit included, it's not rocket science, and trying to make up excuses to go around that is more than stupid.
`>`"the problem with this PR is that it won't accomplish what it intends to do because if people wanted to play sec for the cool loot they fucking would"
Which is why I didn't go the "give sec better equipment" route, giving them a fancy gun isn't going to save them, nor have actual appeal. Giving them innate defense and some combat bonuses? It's a slap in the back to go play security if you lack enough braincells to fuck up in melee combat. And it more than worked, even if people were stupid enough to keep playing due to the test merge being fucked up and not including the officer martial arts, shifts were getting 3-4 officers with a HoS for a while.

also @epicmemename go away tg player looooooooooool

**Comment by ghost** [comment]
they were getting 3-4 officers with a hos because it was a new feature, not something that'll last, flip lasts for 3.5 seconds not for 3.5 deciseconds fix your code, you've missed every single other point and gone back to 'MELEE WEAPONS DON't MATTER BECAUSE GUNS EXIST' and so honestly there's no point continuing this. everything you said is fucking stupid, this PR is fucking stupid, and yes, instant aggro grab IS stunbaton-for-hand.

**Comment by GuppyLaxx** [comment]
I win the argument

**Comment by ghost** [comment]
Convincing the opponent that you are actually cleverbot and so there's no point continuing to talk to you does not count as winning unless you are donald trump

**Comment by ThatLing** [comment]
can we stop posting long messages, i don't have to attention span to read all of these in 1 sitting.

**Comment by MacHac** [comment]
The issue here isn't stuns.  It's not tasers, or the det's gun, or who has the bigger e-peen.  The issues are:
- Does security need a buff?
- Will this change get more players playing security?

Security already shits on every antag it can find, with the exception of crappy officers attacking skilled traitors (who were going to win anyway) and heavily snowballed group antags.  Best case scenario, the new abilities go mostly unused (except for CQC) and security sticks with its tried and true equipment.  Worst case scenario, you've brought the hammer down on solo antags to buy security 2 more minutes of viability in a cult round.

For the second issue, as Adam said, I suspect the sort of players who start playing sec because of free martial arts are the sort of players you really don't want playing sec.  something something stanford prison experiment

Side note: With the exception of the Captain's ability to identify fake nuke disks and the Curator's omnitongue, this would be the first 'innate' buff that any job received.  Innate buffs and abilities encourage powergaming (unstealable, permanent abilities) and metagaming (identifying changelings and the disguised.)  Even if the above two issues were resolved in this PR's favor, I won't support any innate abilities unless they're given some sort of drawback.  For example, the mime can't speak and have his powers at the same time, preventing them from being a straight buff.

**Comment by ghost** [comment]
@MacHac 
I just want to point out that your reference to the Stanford Prison Experiment was an extremely poor one, and bears little to no relevance to the topic at hand.

**Comment by MacHac** [comment]
@Doxdd 
Yes, thank you for @ing me to tell me that you didn't like my reference.  I guess my reference to a famous demonstration of the cruelty of human beings given authority and power over others has little to nothing to do with buffing security.

It especially has nothing to do with attracting new security players with the ability to screw over other players more efficiently and uncounterably than ever before.  No, I must have just been thinking of something else when I made that reference.

The entire security department/antag interaction has always been strongly reminiscent of the SPE, especially during major group antag failures.  I admit that it's not the *most* perfectly placed reference for the topic, but I'm not gonna dig into wikipedia to insult someone on the internet.

**Comment by epicmemename** [comment]
@GuppyLaxx Aggro grabs are very strong considering you can table people from them, and they need to deal with shitty RNG to get out. Flipper is by far the most aggregious, considering it's a 75% chance to stun and it does good damage.

Adam's critque didn't touch upon how when you're playing and you see someone holding a stun baton running at you, you tend to react quickly, because it's telegraphed. This was part of the problem with stun gloves because you have virtually no indicator that they had them, since everyone wore gloves of some kind anyway. Krav Maga being an implant instead of gloves removes the "oh shit it's security with krav maga" alert because before you could at least see red on them. + it removes the decision between insulateds and krav maga and removes actual ingame incentive for the warden to stay in the armoury (one of a kind op gloves) since you can't get them stolen now.

Lets consider an example, you're playing sec and you want to go undercover. You changed into all assistant shit, you have nothing too revealing on you that would make people realize you're carring sec gear on you. You walk around for a bit and then you see the guy who your metafriend on discord said was a traitor. To that traitor, he has no idea you're dangerous to him until you either pull out sec gear, meaning he does have a chance, if he is fast enough, to be get away. If this PR goes through instead of having a chance he'd already be in a 3.5 second stun and then being harmbatoned, with no indicator of you being security. The fact that you make these RNG doesn't even count. Smash Bros Brawl had the infamous random tripping mechanic, and despite only being a 1% chance, it ruined the games chance at actually being competitive.

Okay, this pr is not epic

**Comment by epicmemename** [comment]
Also if you want to increase incentive to playing sec, just do something along the lines of antag rep, which works much better than this for what it's attempting to do.

**Comment by ghost** [comment]
One, no problem.
Two, it really doesn't, you're right.
Three, your whole point is that through giving security martial arts, as well as the HoS CQC, that they will immediately become an absolutely unstoppable gestapo that immediately dispatches any threat with maximum force regardless of the situation. No, security won't immediately start going fucking insane with power just because they have martial arts in some forms. The wardens in the experiment were not policed, the security team is, by the admins. This PR won't change security standards, and people that act out of line in accordance with the rules will be treated as such.
tl;dr you're a moron for making that reference and should be ashamed.
@MacHac

**Comment by MacHac** [comment]
I'm ashamed of myself for taking the time to read your comment rather than doing something more constructive with my time.  The final thing I will say to you is that it's not against game rules to be shitcurity, but that doesn't mean that shitcurity is good for the game.  Powergaming is also bad, and almost impossible to police (especially on yogs.)  Buffing security officers and bribing people to be security with powers will increase both.

**Comment by optimumtact** [comment]
In awe of the size of this lad

**Comment by DreamwalkerSisyphe** [comment]
Nice, I forgot that spawning with tasers needs more of an incentive

**Comment by ThatLing** [comment]
@MacHac SPE was fake lol

**Comment by N3D6** [comment]
Someone reopen this

**Comment by optimumtact** [comment]
I agree, yogs add this :+1:

**Comment by ghost** [comment]
What the hell am I doing here

**Comment by GuppyLaxx** [comment]
i'm banned from yogs i cannot

**Review by Joey139 — APPROVED** [review]
